src/send-message.ts

```diff
@@ -0,0 +1,174 @@
+import { getInput, info, setFailed } from "@actions/core";
+import { Err, Ok, Result } from "ts-results";
+import { AsyncResultWrapper } from "ts-async-results";
+import { type Client, default as zulipInit } from "zulip-js";
+
+type DestinationDetails = PrivateDestination | StreamDestination;
+type PrivateDestination = {
+  kind: DestinationKind.Private;
+  destination: Array<string | number>;
+};
+type StreamDestination = {
+  kind: DestinationKind.Stream;
+  destination: string | number;
+  topic: string;
+};
+
+enum DestinationKind {
+  Private = "private",
+  Stream = "stream",
+}
+
+function assertExhaustiveSwitch(_: never): never {
+  throw new Error("switch statement not exhaustive");
+}
+
+function allNumeric(candidate: string | string[]): boolean {
+  const oneOrMoreNumbersOnly = /^\d+$/;
+  return typeof candidate === "string"
+    ? !!oneOrMoreNumbersOnly.test(candidate)
+    : candidate.every((item: string) => oneOrMoreNumbersOnly.exec(item));
```

Comment:
JS is not my native tongue at all, but `.test` looks to [return a bool](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) (thus not requiring the `!!`) and `.exec` looks to [return a match list](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) (which would need a `!!`)?

Am I missing a reason these aren't both `oneOrMoreNumbersOnly.test(...)` with no `!!` necessary?

Reply:
This might just be an artifact from when I was going through consolidating multiple regex calls into one site: the original JS used `.match`, [which returns a nullable Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) and would have required `!!`. Why and when I moved to `test` without removing the `!!`, I don't recall.

I'm actually somewhat surprised none of the linters picked this up: they complain about variable and type names, but not this redundancy, interesting!